src/Components/ToDo-Components/Task.js

```diff
@@ -3,12 +3,30 @@ import css from "./Task.module.css";
 import PropTypes from "prop-types";
 
 class Task extends Component {
+  statusHandler(completed, id) {
+    this.props.setToDosStatus((prevTodos) => [
+      ...prevTodos.map((todo) => {
+        if (todo.id === id) {
+          todo.completed = completed;
```

Comment:
Return a new object instead of directly changing the value.
`return {...todo, completed: !todo.completed}`